crates/sui-framework/src/lib.rs

```diff
@@ -12,115 +12,93 @@ use sui_types::{
     error::SuiResult,
     move_package::MovePackage,
     object::{Object, OBJECT_START_VERSION},
-    MOVE_STDLIB_OBJECT_ID,
+    MOVE_STDLIB_ADDRESS, SUI_FRAMEWORK_ADDRESS,
 };
 
 pub mod natives;
 
-static SUI_FRAMEWORK_BYTES: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/sui-framework"));
-static SUI_FRAMEWORK: Lazy<Vec<CompiledModule>> = Lazy::new(|| {
-    get_sui_framework_bytes()
-        .into_iter()
-        .map(|module| CompiledModule::deserialize(&module).unwrap())
-        .collect()
-});
-
-static SUI_FRAMEWORK_TEST: Lazy<Vec<CompiledModule>> = Lazy::new(|| {
-    const SUI_FRAMEWORK_TEST_BYTES: &[u8] =
-        include_bytes!(concat!(env!("OUT_DIR"), "/sui-framework-test"));
-
-    let serialized_modules: Vec<Vec<u8>> = bcs::from_bytes(SUI_FRAMEWORK_TEST_BYTES).unwrap();
-
-    serialized_modules
-        .into_iter()
-        .map(|module| CompiledModule::deserialize(&module).unwrap())
-        .collect()
-});
-
-static MOVE_STDLIB_BYTES: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/move-stdlib"));
-static MOVE_STDLIB: Lazy<Vec<CompiledModule>> = Lazy::new(|| {
-    get_move_stdlib_bytes()
-        .into_iter()
-        .map(|module| CompiledModule::deserialize(&module).unwrap())
-        .collect()
-});
-
-static MOVE_STDLIB_TEST: Lazy<Vec<CompiledModule>> = Lazy::new(|| {
-    const MOVE_STDLIB_TEST_BYTES: &[u8] =
-        include_bytes!(concat!(env!("OUT_DIR"), "/move-stdlib-test"));
-
-    let serialized_modules: Vec<Vec<u8>> = bcs::from_bytes(MOVE_STDLIB_TEST_BYTES).unwrap();
-
-    serialized_modules
-        .into_iter()
-        .map(|module| CompiledModule::deserialize(&module).unwrap())
-        .collect()
-});
-
-pub fn get_sui_framework() -> Vec<CompiledModule> {
-    Lazy::force(&SUI_FRAMEWORK).to_owned()
+macro_rules! system_package {
+    ($address:expr, $Package:ident, $path:literal, [$($Dep:ident),* $(,)?]) => {
+        pub struct $Package;
+
+        impl SystemPackage for $Package {
+            const ID: ObjectID = ObjectID::from_address($address);
+            const BCS_BYTES: &'static [u8] = include_bytes!(concat!(env!("OUT_DIR"), "/", $path));
```

Comment:
Just wanted to call out that this is assuming the correct version of the system package has already been written out to this location at compile time.

Reply:
Yes, this is already handled by the `sui-framework-build` crate (this is pre-existing).